### DISCUSSION TITLE
TCP and/or UDP

### DISCUSSION THREAD

**Discussion by Geffers** [pivpn/pivpn#1868] [Support]
Folks, does the PiVPN server have to operate in either UDP or TCP or can it support both protocols at the same time?

I appreciate the client config file needs changing to one or the other.

Also, if using TCP can port be left as 1194 or is it preferable to use 443?

Geffers

**Comment by orazioedoardo** [comment]
Running OpenVPN on both UDP as well as TCP is possible but not supported via PiVPN. What you could do is run WireGuard on any port (UDP) and OpenVPN at the same time on TCP 443 to get the best chance of connectivity on limited networks. This is supported via PiVPN.

**Comment by Geffers** [comment]
I thought with PiVPN you had to choose wire guard or openVPN at
installation time, didn't realise both could run at same time or am I
misunderstanding and it is two PiVPN installations?

Geoff

**Reply by orazioedoardo** [reply]
You choose one, then run the install script again and choose the other.
